storage/persistence/etcd/etcd.go

```diff
@@ -26,21 +26,37 @@ import (
 	"sync"
 	"time"
 
-	"github.com/RocksLabs/kvrocks_controller/logger"
-	"github.com/RocksLabs/kvrocks_controller/metadata"
-	"github.com/RocksLabs/kvrocks_controller/storage/persistence"
-
 	clientv3 "go.etcd.io/etcd/client/v3"
 	"go.etcd.io/etcd/client/v3/concurrency"
+	"go.etcd.io/etcd/pkg/transport"
 	"go.uber.org/atomic"
 	"go.uber.org/zap"
+
+	"github.com/RocksLabs/kvrocks_controller/logger"
+	"github.com/RocksLabs/kvrocks_controller/metadata"
+	"github.com/RocksLabs/kvrocks_controller/storage/persistence"
 )
 
 const (
 	sessionTTL         = 6
 	defaultDailTimeout = 5 * time.Second
 )
 
+type Config struct {
+	Addrs     []string `yaml:"addrs"`
+	BasicAuth struct {
+		Enable   bool   `yaml:"enable"`
+		Username string `yaml:"username"`
+		Password string `yaml:"password"`
+	} `yaml:"basic_auth"`
```

Comment:
```suggestion
    Username string `yaml:"username"`
	Password string `yaml:"password"`
```

The etcd client will use the auth configuration only if the username and password are not empty: https://github.com/etcd-io/etcd/blob/main/client/v3/client.go#L388-L392. So I think we can assign the username/password to etcd.Client directly.

Reply:
done 😁